### PR TITLE
Use Infinispan 13 server for integration tests

### LINKED ISSUE #3010: Use Infinispan 13 for running integration tests
There won't be much work going into Infinispan 11 anymore so upgrading to the latest major version seems reasonable as that will be the version that will be used in most production environments anyway.

### PR DESCRIPTION
**Pull request by sophokles73**
Infinispan 13 does not support ECC based certificates (yet). Thus, the
Infinispan server's configuration has been changed to no longer use TLS.
In a production environment, the server can still be configured with an
RSA based key pair.

The cache configuration has been changed to a simple in-memory cache
because there is not much use for being able to restore cache state from
the file system when running integration tests.

Fixes #3010

### CODE REVIEW

**Comment by sophokles73** [comment]
@calohmn would you mind taking a look?

File: tests/src/test/resources/infinispan/hono-data-grid.xml [hunk 1]
```diff
@@ -1,43 +1,33 @@
...
       </security>
     </local-cache>
   </cache-container>
 
-  <server xmlns="urn:infinispan:server:11.0">
+  <server xmlns="urn:infinispan:server:13.0">
     <interfaces>
       <interface name="public">
-        <non-loopback />
+        <server:inet-address value="0.0.0.0" />
```
Comment (calohmn) [inline tests/src/test/resources/infinispan/hono-data-grid.xml:30]: nit: The `server` prefix looks superfluous here (`<server>` tree already having the corresponding namespace).

**Comment by sophokles73** [comment]
@calohmn I have pushed a fix

**Review by calohmn — APPROVED** [review]
LGTM

**Pull request merged**